client/src/app/auth/reset-password/page.tsx

```diff
@@ -0,0 +1,167 @@
+"use client";
+import { useState, useEffect } from "react";
+import { useRouter, useSearchParams } from "next/navigation";
+import useFetch from "@/src/hooks/useFetch";
+import { AuthService } from "@/src/services/AuthService";
+import Navbar from "@/src/components/Navbar/Navbar";
+import Footer from "@/src/components/Footer/Footer";
+import useThemeStore from "@/src/store/themeStore";
+import Image from "next/image";
+import ResetPasswordSvg from "@/public/assets/Login.svg";
+import Link from "next/link";
+
+const ResetPassword = () => {
+  const router = useRouter();
+  const searchParams = useSearchParams();
+  const theme = useThemeStore((state) => state.theme);
+  const [token, setToken] = useState<string | null>(null);
+  const [password, setPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+  const [successMessage, setSuccessMessage] = useState("");
+
+  const [{ isLoading, data, isError, error }, resetPasswordAPI] = useFetch(null);
+
+  useEffect(() => {
+    const token = searchParams.get("token");
+    if (token) {
+      setToken(token);
+    }
+  }, [searchParams]);
+
+  useEffect(() => {
+    if (data) {
+      const { code, message } = data;
+      if (code === 200) {
+        setSuccessMessage("Password has been reset successfully");
+        router.push("/auth/login");
+      } else {
+        setErrorMessage(message);
+      }
+    } else if (isError) {
+      setErrorMessage(error.message);
+    }
+  }, [data, isError, error, router]);
+
+  const onHandleSubmit = () => {
+    if (!password) {
+      setErrorMessage("Enter Password");
+    } else if (!confirmPassword) {
+      setErrorMessage("Enter Confirm Password");
+    } else if (password !== confirmPassword) {
+      setErrorMessage("Passwords do not match");
+    } else if (!token) {
+      setErrorMessage("Invalid token");
```

Comment:
We are not verifying token here, we cannot decide token is valid or not, just say token is missing.